### PR TITLE
meta/autoid: fix AUTO_ID_CACHE 1 setting affect row id allocator performance

### LINKED ISSUE #39528: autoid cache 1 impact the row id allocation performance
## Enhancement

When I'm testing the performance of https://github.com/pingcap/tidb/issues/38442

[autoid.txt](https://github.com/pingcap/tidb/files/10129197/autoid.txt)

I use this table schema:

```
create table test (id int unsigned key nonclustered auto_increment) shard_row_id_bits=4 auto_id_cache 1;
```

It avoids the sequent rowid caused tikv hotspot issue.
But I find the flamegraph abnormal:

image

Row ID allocating takes too much resource!
I figure out that cause by the side effect of the auto_id_cache = 1 setting.

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #39528

Problem Summary:

### What is changed and how it works?

When AUTO_ID_CACHE is set to 1, let the row id allocator still use the default cache size (30000).

### Check List

Tests 

- [x] Unit test

On my local machine, before the fix, QPS is 7k, after fix, QPS is 40k

- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -3036,21 +3036,6 @@ func TestAutoIncrementForceAutoIDCache(t *testing.T) {
...
-	// Cannot set next global ID to 0.
-	tk.MustExec("alter table t force auto_increment = 0;")
-	tk.MustExec("alter table t force auto_increment = 1;")
-	require.Equal(t, uint64(3), getNextGlobalID())
-	// inserting new rows can overwrite the existing data.
-	tk.MustExec("insert into t values (3);")
-	tk.MustExec("insert into t values (3);")
-	tk.MustQuery("select a, _tidb_rowid from t;").Check(testkit.Rows("1 1", "2 2", "3 3", "3 4"))
-	tk.MustExec("drop table if exists t;")
-
```
Comment (tiancaiamao) [inline ddl/db_integration_test.go:3053]: This code is duplicated with the following lines, it's a mistake when copying the code

File: ddl/db_integration_test.go [hunk 2]
```diff
@@ -3060,11 +3045,14 @@ func TestAutoIncrementForceAutoIDCache(t *testing.T) {
 	// Cannot set next global ID to 0.
 	tk.MustExec("alter table t force auto_increment = 0;")
 	tk.MustExec("alter table t force auto_increment = 1;")
-	require.Equal(t, uint64(3), getNextGlobalID())
+	tk.MustQuery("show table t next_row_id").Check(testkit.Rows(
```
Comment (tiancaiamao) [inline ddl/db_integration_test.go:3048]: when auto_id_cache=1, row id cache is still used, so it's not 3, instead, it's a bigger number

**Pull request merged**